rust/operator-binary/src/backend/k8s_search.rs

```diff
@@ -133,16 +141,21 @@ fn build_label_selector_query(
     vol_selector: &SecretVolumeSelector,
     pod_info: LabelSelectorPodInfo,
 ) -> Result<String, Error> {
-    let mut label_selector =
-        BTreeMap::from([(LABEL_CLASS.to_string(), vol_selector.class.to_string())]);
+    let mut labels: Labels =
+        BTreeMap::from([(LABEL_CLASS.to_string(), vol_selector.class.to_string())])
+            .try_into()
+            .context(BuildLabelSnafu)?;
     let mut listener_i = 0;
     // Only include node selector once we are scheduled,
     // until then we use the query to decide where scheduling should be possible!
     if let LabelSelectorPodInfo::Scheduled(pod_info) = pod_info {
         // k8sSearch doesn't take the scope's resolved addresses into account, so we need to check whether
         // Listener scopes also imply Node
         if pod_info.scheduling.has_node_scope {
-            label_selector.insert(LABEL_SCOPE_NODE.to_string(), pod_info.node_name.clone());
+            labels.insert(
+                Label::try_from((LABEL_SCOPE_NODE.to_string(), pod_info.node_name.clone()))
+                    .context(BuildLabelSnafu)?,
+            );
```

Comment:
suggestion: Version `0.61.0` of operator-rs introduces the `parse_insert` function. This could be shortened to:

```suggestion
            labels
                .parse_insert((LABEL_SCOPE_NODE.to_string(), pod_info.node_name.clone()))
                .context(BuildLabelSnafu)?;
```

Reply:
The same could be applied to other inserts below.

Reply:
resolved in 3187842